### PR TITLE
Fix water bug in packed octree

### PR DESCRIPTION
**Pull request by aTom3333**
This PR fixes the bug causing water not to look as expected when using the packed octree

expected result:
water-node-octree

result with buggy packed octree:
water-packed-octree

This being said, i believe this PR needs a bit a discussion before being merged.
As i've explain earlier on discord, this issue does not occur with the node based octree because, when testing if children must be merged, they all are compared against the old node that has just been replaced that prevents the merging from happening because the children, even if they all compare equals don't compare equals to the old node.
I believe this behavior is a bug but i think its only bad consequence is that sometimes nodes are not merged when they could have been, wasting a bit of memory.

The bug that i think is really the cause of the water problem is the fact that when merging nodes, the data is not propagated to the parent. This is the bug fixed by this PR (only for the packed octree).
This bug is present in the node based octree but its effect wasn't noticeable because of the other bug that prevents merging.

To recap, from my understanding, there are 3 bugs, 2 being the same bug in two octree implementation. The bug in packed octree was noticeable, but the 2 bugs in nodes based octree weren't cause they are "cancelling" each other.

I think we should fix the 2 bugs in the node based octree as well but i'd like to wait for feedback to see if my interpretation is not wrong.

### CODE REVIEW

**Comment by leMaik** [comment]
> The bug that i think is really the cause of the water problem is the fact that when merging nodes, the data is not propagated to the parent.

@aTom3333 I agree that this is what should be fixed, for both octree implementations. Also, the `data` needs to be checked before merging the nodes so that only nodes with identical type and data get merged.

**Comment by aTom3333** [comment]
The data is checked for both implementation prior to the merge

**Comment by leMaik** [comment]
Sounds good to me then :100:

**Comment by aTom3333** [comment]
So i just fixed the 2 bugs i talked about in the node based octree, unfortunately, in order to propagate data to the parent node, the parent node needs to be replaced by a DataNode, so we need to keep track of the grandparent node and the index of the parent in the grand parent. In theory it could even bubble up to the root that might have to be replaced and is no longer final

**Comment by leMaik** [comment]
Nice! :tada:
